File: 0283-move-zeroes/0283-move-zeroes.java
class Solution {
    public void moveZeroes(int[] nums) {
        int zeroSize = 0;
        for (int i = 0; i < nums.length; i++) {
            if (nums[i] == 0) {
                zeroSize++;
            } else {
                int temp = nums[i];
                nums[i] = 0;
                nums[i - zeroSize] = temp;
            }
        }
    }
}

File: 0283-move-zeroes/0283-move-zeroes.kt
class Solution {
fun moveZeroes(nums: IntArray): Unit {
    var index = 0
    for(i in nums.indices) {
        if(nums[i] != 0) {
            nums[index++] = nums[i]
        }
    }

    while (index < nums.size) {
        nums[index++] = 0
    }

}
}